Add `Always show pre-publish checks` checkbox

octaedro opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

The sidebar/modal can be optional when the form doesn't contain any validation error.
This PR adds the `Always show pre-publish checks` checkbox.

![Screenshot 2024-02-14 at 10 14 43](https://github.com/woocommerce/woocommerce/assets/1314156/f263ea67-fdb7-4bc6-8954-d4d562563334)

Closes #43707.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the New product editor and the Prepublish sidebar (product-pre-publish-modal) are enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add New.
3. Add a product name and press `Publish`.
4. The prepublish panel should be visible.
5. Verify that the `Always show pre-publish checks` checkbox is visible.

![Screenshot 2024-02-14 at 16 07 23](https://github.com/woocommerce/woocommerce/assets/1314156/a8775f9c-6f30-473e-9ac7-1acce5be164d)

6. It should be checked by default.
7. Uncheck the checkbox and close the panel.
8. Press Publish again. Now, the panel shouldn't be visible anymore.
9. If you want to re-enable the panel, you should remove the option `woocommerce_show_prepublish_checks_enabled` from the database. You can do it at Tools > WCA Test Helper > Options.
10. Also verify the checkbox looks good on different screen sizes.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

octaedro committed:
Add show-prepublish-checks-section

# Conflicts:
#	packages/js/product-editor/src/components/prepublish-panel/prepublish-panel.tsx

# Conflicts:
#	packages/js/product-editor/src/components/prepublish-panel/style.scss

octaedro committed:
Add show-prepublish-checks hook

octaedro committed:
Refactor prepublish panel button visiblity

# Conflicts:
#	packages/js/product-editor/src/components/header/header.tsx

# Conflicts:
#	packages/js/product-editor/src/components/header/header.tsx

octaedro committed:
Fix footer styles

octaedro committed:
Add changelogs

octaedro committed:
Add content div

octaedro committed:
Fix styles

# Conflicts:
#	packages/js/product-editor/src/components/prepublish-panel/style.scss

octaedro committed:
Rename recorded action

octaedro committed:
Fix styles

octaedro committed:
Hide panel after unchecking option

octaedro committed:
Improve resolving

mdperez86 reviewed (approved):
Nice job @octaedro LGTM!!!

veljkho committed:
Add new E2E test for merchant transforming classic cart to cart block (#44926)

jonathansadowski committed:
Update the monorepo-utils gitignore to only include index.js and its license file. (#44932)

ObliviousHarmony committed:
Fixed `@woocommerce/plugin-woocommerce` Watch (#44930)

By default `nodemon` only watches for JS file changes. This expands the
config so that it watch the file extensions that are actually
produced by builds.

jonathansadowski committed:
Update core package.json to remove nightly ci config (#45096)

* Update core package.json to remove nightly ci config

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

roykho committed:
Update blocks build message (#44811)

* Update blocks build message

* Add changefile(s) from automation for the following project(s): woocommerce

* Update plugins/woocommerce/src/Blocks/Domain/Bootstrap.php

Co-authored-by: Christopher Allford <6451942+ObliviousHarmony@users.noreply.github.com>

* Update plugins/woocommerce/src/Blocks/Domain/Bootstrap.php

Co-authored-by: Christopher Allford <6451942+ObliviousHarmony@users.noreply.github.com>

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Christopher Allford <6451942+ObliviousHarmony@users.noreply.github.com>

nefeline committed:
[CYS on Core] Update the themes list on the intro screen (#44822)

* Update themes list on core. Update the themes data and remove hardcoded reference to free themes.

* Update styles for the Free, Paid and active cards.

* Add the color palletes to TT4 theme.

* Update the references for default and core themes.

* Add changefile(s) from automation for the following project(s): woocommerce

* Update the link to TT4 theme

* Update default value for price.

---------

Co-authored-by: github-actions <github-actions@github.com>
